src/components/topbar/TopBar.jsx

```diff
@@ -0,0 +1,54 @@
+import "./topbar.css";
+import { Search, Person, Chat, Notifications } from "@material-ui/icons";
+import { Link, useHistory } from "react-router-dom";
+import { withAuth } from "../../HOC/withAuth";
+
+const TopBarComponent = ({ user, setUser }) => {
+  const history = useHistory();
+  const handleSubmit = () => {
```

Comment:
avoid generic names, function names should be helpful enough to tell us what function does.